Microsoft.Toolkit.Uwp/Helpers/BluetoothLEHelper/BluetoothLEHelper.cs

```diff
@@ -0,0 +1,350 @@
+
+        /// <summary>
+        /// Gets a value indicating whether central role is supported by this device
+        /// </summary>
+        public bool IsCentralRoleSupported => _adapter.IsCentralRoleSupported;
+
+        /// <summary>
+        /// Gets a value indicating whether the Bluetooth LE Helper is supported.
+        /// </summary>
+        public bool IsBluetoothLeSupported => ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 4);
+
+        /// <summary>
+        /// An event for when the enumeration is complete.
+        /// </summary>
+        public event EventHandler<EventArgs> EnumerationCompleted;
+
+        /// <summary>
+        /// Starts enumeration of bluetooth device
+        /// </summary>
+        public void StartEnumeration()
+        {
+            if (_advertisementWatcher?.Status == BluetoothLEAdvertisementWatcherStatus.Started)
+            {
+                return;
+            }
+
+            // Additional properties we would like about the device.
+            string[] requestedProperties =
+            {
+                "System.Devices.Aep.Category",
+                "System.Devices.Aep.ContainerId",
+                "System.Devices.Aep.DeviceAddress",
+                "System.Devices.Aep.IsConnected",
+                "System.Devices.Aep.IsPaired",
+                "System.Devices.Aep.IsPresent",
+                "System.Devices.Aep.ProtocolId",
+                "System.Devices.Aep.Bluetooth.Le.IsConnectable"
+            };
+
+            // BT_Code: Currently Bluetooth APIs don't provide a selector to get ALL devices that are both paired and non-paired.
+            _deviceWatcher = DeviceInformation.CreateWatcher(
+                BluetoothLeDeviceWatcherAqs,
+                requestedProperties,
+                DeviceInformationKind.AssociationEndpoint);
+
+            // Register event handlers before starting the watcher.
+            _deviceWatcher.Added += DeviceWatcher_Added;
+            _deviceWatcher.Updated += DeviceWatcher_Updated;
+            _deviceWatcher.Removed += DeviceWatcher_Removed;
+            _deviceWatcher.EnumerationCompleted += _deviceWatcher_EnumerationCompleted;
+
+            _advertisementWatcher = new BluetoothLEAdvertisementWatcher();
+            _advertisementWatcher.Received += AdvertisementWatcher_Received;
+
+            BluetoothLeDevices.Clear();
+
+            _deviceWatcher.Start();
+            _advertisementWatcher.Start();
+        }
+
+        /// <summary>
+        /// Stops enumeration of bluetooth device
+        /// </summary>
+        public void StopEnumeration()
+        {
+            if (_deviceWatcher != null)
+            {
+                _deviceWatcher.Added -= DeviceWatcher_Added;
+                _deviceWatcher.Updated -= DeviceWatcher_Updated;
+                _deviceWatcher.Removed -= DeviceWatcher_Removed;
+
+                if (EnumerationCompleted != null)
```

Comment:
Why checking if the event is subscribed here when it's not being checked above on line 152?